Source/SpectrumAnalysis/KTSeqTrackFinder.cc

```diff
@@ -0,0 +1,539 @@
+/*
+ * KTSeqTrackFinder.cc
+ *
+ *  Created on: Sep 15, 2016
+ *      Author: Christine
+ */
+
+#include "KTSeqTrackFinder.hh"
+#include "KTLogger.hh"
+
+
+#include <numeric>
+#include <cmath>
+
+using std::list;
+using std::set;
+using std::vector;
+
+
+namespace Katydid
+{
+    KTLOGGER(stflog, "DansTrackFinder");
+
+
+    KT_REGISTER_PROCESSOR(KTSeqTrackFinder, "seq-clustering");
+
+    KTSeqTrackFinder::KTSeqTrackFinder(const std::string& name) :
+            KTProcessor(name),
+            fMode(eSNR_Power),
+            fTrimmingFactor(19.8),
+            fLinePowerWidth(4),
+            fPointAmplitudeAfterVisit(0),
+            fMinFreqBinDistance(10),
+            fTimeGapTolerance(0.5*pow(10,-3)),
+            fFrequencyAcceptance(56166.0528183),
+            fSearchRadius(6),
+            fConvergeDelta(1.5),
+            fSNRPowerThreshold(6.),
+            fMinPoints(2),
+            fMinSlope(0.0),
+            fInitialSlope(3.0*pow(10,8)),
+            fMinBin(0),
+            fMaxBin(1),
+            fBinWidth(0.0),
+            fMinFrequency(0.),
+            fMaxFrequency(1.),
+            fCalculateMinBin(true),
+            fCalculateMaxBin(true),
+            fActiveLines(),
+            fNLines(0),
+            fApplyPowerCut(false),
+            fApplyDensityCut(false),
+            fPowerThreshold(0.0),
+            fDensityThreshold(0.0),
+            fTrackSignal("pre-candidate", this),
+            fClusterDoneSignal("clustering-done", this),
+            fGainVarSlot("gv", this, &KTSeqTrackFinder::SetPreCalcGainVar),
+            //fPSPreCalcSlot("ps-pre", this, &KTSeqTrack::PointLineAssignment),
+            fPSSlot("ps-in", this, &KTSeqTrackFinder::RunSequentialTrackFinding),
+            fDoneSlot("done", this, &KTSeqTrackFinder::AcquisitionIsOver, &fClusterDoneSignal)
+    {
+    }
+
+    KTSeqTrackFinder::~KTSeqTrackFinder()
+    {
+    }
+
+    bool KTSeqTrackFinder::Configure(const scarab::param_node* node)
```

Comment:
Two comments here:

1. There are some hard tabs in here that need to be converted to 4 spaces each.

1. In most cases you can use this pattern for setting config values:
    ```C++
    SetMyVariable(node->get_value("my-variable", GetMyVariable());
    ```
    In this case, we're always calling the Set and Get functions, but that's usually ok since they're inline functions call and this is a non-time-critical part of the operation.

    That's much more concise and simpler to maintain than
    ```C++
    if(node->has("my-variable"))
    {
        SetMyVariable(node->get_value("my-variable"));
    }
    ```
    The exceptions to that would be the situation where you explicitly don't want to call the Set function unless the variable is there.